### DISCUSSION TITLE
Pull request opened by edit target to my forked repo instead of origin repo

### DISCUSSION THREAD

**Discussion by jjshao-ms** [community/community#62973] [Pull Requests]
### Select Topic Area

Bug

## Repro steps
1. Goto [this page](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/index.md)
2. Click the edit button
  <img width="244" alt="image" src="https://github.com/community/community/assets/87108022/27448562-4236-44a9-8d1e-82733ef5b0a6">

3. Fork the repo as it asked.
4. Make some change
5. Click "**Commit changes**" button
6. Click "**Propose changes**" with the default new branch name.
<img width="365" alt="image" src="https://github.com/community/community/assets/87108022/0fc6d7ff-77fd-4791-b509-60e1b60e7251">

7. The creating PR/compare page has a base branch that is main branch of my forked repo instead of the origin(dotnet/AspNetCore.Docs) repo's main branch.
<img width="941" alt="image" src="https://github.com/community/community/assets/87108022/8631ab56-955b-4a79-a45f-b4027f0fd4b6">

## Expect
In step 7, I should see base branch to be `dotnet/AspNetCore.Docs`, like this when I create the PR from [Pull request page](https://github.com/dotnet/AspNetCore.Docs/pulls)
<img width="938" alt="image" src="https://github.com/community/community/assets/87108022/68fdc387-4b9e-4ee0-95a9-2f4e7a1b1abd">

## When does it work before
Still works last week.

**Comment by nickwalkmsft** [comment]
This change was also observed in https://github.com/orgs/community/discussions/11729#discussioncomment-6631489, where a lot of folks were asking for it to work this way.

**Comment by jjshao-ms** [comment]
Personally, I would expect this PR target to original repo, because this edit button was mainly for one time update (you can only change one file within edit button editor, that is a more one-time thing). I won't be happy if I need to open two PR or manually change the base path for such one time, one file PR.

**Reply by jandsonrj** [reply]
I understand your perspective. I agree that, given the context of the one-time update provided by the edit button, it would be more convenient to target the pull request to the original repository. This would help maintain workflow integrity by avoiding the need to open two PRs or manually adjust the base path for a single file. I believe this approach would streamline the process and contribute to a more efficient change management experience.

**Comment by per1234** [comment]
Something to note is that the default user experience is much different depending on whether the ["Require a pull request before merging"](https://docs.github.com/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/available-rules-for-rulesets#require-a-pull-request-before-merging) rule is enabled in the target repo's ["branch protection rules"](https://docs.github.com/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches).

## Default Workflow when Branch Protection Rule Is Not Enabled

1. Navigate in the target repository to the file you want to propose a change to.
1. Click the pencil icon.
   A "**You need to fork this repository to propose changes.** page opens.
1. Click the "**Fork this repository**" button in the "**You need to fork this repository to propose changes.** page.
   The file opens in the online editor.
1. Edit the file in the online editor.
1. Click the "**Commit changes...**" button.
   A "**Commit changes**" dialog opens.
   **ⓘ** Note that the radio button next to "**Commit directly to the \<target branch\> branch**" is selected by default.
1. Click the "**Commit changes**" button in the "**Commit changes**" dialog.
   A "**Comparing changes**" page opens.
   **ⓘ** Note that the page has a "**base repository**" drop-down menu and that the target repository is selected by default.
1. Click the "**Create pull request**" button in the "**Comparing changes**" page.
   An "**Open a pull request**" page opens.
1. Click the "**Create pull request**" button in the "**Open a pull request**" page.

🙂 The user's contribution was submitted to the target repository as they intended.

## Default Workflow when Branch Protection Rule Is Enabled

1. Navigate in the target repository to the file you want to propose a change to.
1. Click the pencil icon.
   A "**You need to fork this repository to propose changes.** page opens.
1. Click the "**Fork this repository**" button in the "**You need to fork this repository to propose changes.** page.
   The file opens in the online editor.
1. Edit the file in the online editor.
1. Click the "**Commit changes...**" button.
   A "**Propose changes**" dialog opens.
   **ⓘ** Note that the dialog contains the message:
   > **You can’t commit to \<target branch\> because it is a protected branch**

   **ⓘ** Note that radio button next to "**Create a new branch for this commit and start a pull request**" is selected by default (and is the only option).
1. Click the "**Propose changes**" button in the "**Propose changes**" dialog.
   A "**Comparing changes**" page opens.
   **ⓘ** Note that the page _does not_ have a "**base repository**" drop-down menu, meaning that the user is forced to use their fork as the base repository.
1. Click the "**Create pull request**" button in the "**Comparing changes**" page.
   An "**Open a pull request**" page opens.
   **ⓘ** Note that the page _does not_ have a "**base repository**" drop-down menu, meaning that the user is forced to use their fork as the base repository.
1. Click the "**Create pull request**" button in the "**Open a pull request**" page.

🙁 The user's contribution was submitted to their fork. The maintainers of the target repository will never see it. The user will wonder why their contribution was ignored and be left with a negative impression of contributing to open source software projects.

🐛 The claim in the "**Propose changes**" dialog that committing to the target branch is impossible "because it is a protected branch" is false because, even though the target repo does have that rule enabled, forks don't inherit branch protection settings from the parent. The user is committing to the branch in their fork, which does not have the rule enabled. I submitted a bug report about this to GitHub.

## Required Workflow when Branch Protection Rule Is Enabled

1. Navigate to the home page of the target repository.
1. Click the "**Fork**" button.
   A "**Create a new fork**" page opens.
1. Click the "**Create fork**" button in the "**Create a new fork**" page.
1. Wait for the forking process to finish.
1. Navigate to the file you want to propose a change to.
1. Click the pencil icon.
   The file opens in the online editor.
1. Edit the file in the online editor.
1. Click the "**Commit changes...**" button.
   A "**Commit changes**" dialog opens.
   **ⓘ** Note that the radio button next to "**Commit directly to the \<target branch\> branch**" is selected by default.
1. Click the "**Commit changes**" button in the "**Commit changes**" dialog.
   The page of the edited file opens in the repository.
   😕 Unlike the other workflows, the user flow halts here, leaving no intuitive path to the next step in the contribution workflow.
1. Navigate to the home page of the fork repository.
1. Click the "**Contribute**" text.
   😕 It isn't styled as a button.
   A "Open a pull request to contribute your changes upstream." ~dialog opens.
1. Click the "**Open pull request**" button in the "Open a pull request to contribute your changes upstream." ~dialog.
   An "**Open a pull request**" page opens.
   **ⓘ** Note that the page has a "**base repository**" drop-down menu and that the target repository is selected by default.
1. Click the "**Create pull request**" button in the "**Open a pull request**" page.

😕 The user flow was significantly less intuitive.
